src/librustc_unicode/char.rs

```diff
@@ -234,22 +299,63 @@ impl char {
     #[inline]
     pub fn escape_default(self) -> EscapeDefault { C::escape_default(self) }
 
-    /// Returns the number of bytes this character would need if encoded in
-    /// UTF-8.
+    /// Returns the number of bytes the character this `char` would need if encoded in UTF-8.
+    ///
+    /// The `char` type is always four bytes, but not every character needs the full four when
+    /// encoded as UTF-8. `len_utf8` returns this length.
```

Comment:
The UTF-8 encoding of a code point above U+FFFF (which takes four bytes) and the memory representation of a `char` for that code point are not the same. (UTF-8 uses bit patterns such that you can tell if any give bytes starts a new code point.) That they are both four bytes long is (almost) a coincidence.

I suggest replacing this paragraph with:
> That number of bytes is always between 1 and 4, inclusive.

… and not mentioning `size_of::<char>()` at all.